src/Chapter2.hs

```diff
@@ -842,7 +884,11 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+
+rotate :: Int -> [a] -> [a]
+rotate r xs
+  | r < 0 = []
+  | otherwise = take (length xs) (drop r (cycle xs))
```

Comment:
Awesome! ⭐ 

Another optimisation could be to use `mod` by the list length on `r`, so it won't do unnecessary work 🙂